src/index.js

```diff
@@ -50,27 +53,40 @@ async function serviceembed(params, url, log) {
   }, {});
   // add the URL
   qs.url = url;
-  if (params.OEMBED_RESOLVER_PARAM && params.OEMBED_RESOLVER_KEY && await isWithinRange(params.__ow_headers['x-forwarded-for'], params.WHITELISTED_IPS)) {
+  const api = new URL(params.api || params.OEMBED_RESOLVER_URI);
+  if (params.OEMBED_RESOLVER_PARAM && params.OEMBED_RESOLVER_KEY
+  && await isWithinRange(params.__ow_headers['x-forwarded-for'], await ipList, params.WHITELISTED_IPS)) {
     qs[params.OEMBED_RESOLVER_PARAM] = params.OEMBED_RESOLVER_KEY;
+    Object.entries(qs).forEach(([k, v]) => {
```

Comment:
why do you only copy the `qs` inside this if?

Reply:
I don't understand this question?

Reply:
so before it was like this:
```js
  // add the URL
  qs.url = url;
  if (params.OEMBED_RESOLVER_PARAM && params.OEMBED_RESOLVER_KEY && await isWithinRange(params.__ow_headers['x-forwarded-for'], params.WHITELISTED_IPS)) {
    qs[params.OEMBED_RESOLVER_PARAM] = params.OEMBED_RESOLVER_KEY;
    log.info(`Using embedding service ${params.api || params.OEMBED_RESOLVER_URI} for URL ${url}`);
  }
  return request({
    uri: params.api || params.OEMBED_RESOLVER_URI,
    qs,
    json: true,
  }).then((json) => ({
```

so the request included all the params.

but AFAICS, with your change, the `Object.entries()` is inside the `if`, so it only adds the params from `qs` to `api`, if the condition is true.

Reply:
You make a good point, but why send the request at all if the IP isn't white listed?

@trieloff since you're up at this hour?

Reply:
Yup; you're right again. Jeez, thanks @tripodsan for your quick eye!